es6/README.md

```diff
@@ -0,0 +1,403 @@
+
+- `eslint-plugin-import`
+- `eslint-import-resolver-webpack`
+
+The lint configuration is the definitive source for rules; this document explains the most notable ones. `eslint <source-files> --fix` will fix most formatting issues, such as spacing. 
+
+If React/JSX is in use, use the React default configuration `mobify-es6-react.yml`, which also requires the `eslint-plugin-react` module.
+
+## Overriding Lint Rules
+
+Some of the lint rules disallow uncommon but valid behaviour that is easily confused with/typoed from much more common behaviour. If you need to use the disallowed behaviour on purpose, use an explicit lint override in the source.
+
+For example, the `no-bitwise` rule disallows the bitwise operators `&` and `|` due to their unpredictable (but valid!) behaviour when substituted for the logical operators `&&` and `||`. To use bitwise operations in a situation that's approprate, do the following:
+
+```javascript
+// Mask off the bottom four bits of the octet
+/* eslint-disable no-bitwise */
+let maskedValue = rawValue & 0xf0
+/* eslint-enable no-bitwise */
+```
+
+## Removing Lint Rules
+
+These rules are extensive and a work in progress. If you find any of the rules to be a hindrance in your own work, open an issue in the `mobify-code-style` repository to discuss its removal or modification.
+
+# Base Language
+
+## Always use curly brackets with control statements
+```javascript
+//bad!
+if (x > y)
+    return x - y
+else
+    return y - x
+    
+for (let item of cartItems) 
+    console.log(item)
+    
+//good!
+if (x > y) {
+    return x - y
+} else {
+    return y - x
+}
+    
+for (let item of cartItems) {
+    console.log(item)
+}
+   
+```
+
+## Avoid 'Yoda' conditions
+Conditions read better with the variable on the left and the constant on the right.
+
+```javascript
+// Bad
+if (null !== x && 5 < y) {
+    do(something)
+}
+
+// Good
+if (x !== null && y > 5) {
+    do(something)
+}
+```
+
+## Use object shorthand when possible
+ES6 allows us to not repeat ourselves when including a name from the current scope as an object property.
+
+```javascript
+// Bad
+return {
+    name: name,
+    address: address,
+    phone: phone,
+}
+
+// Good
+return {
+   name,
+   address,
+   phone
+}
+```
+
+## Prefer dot notation
+
+It is clearer to use dot notation instead of bracket notation when it is available (i.e. when the property name is a constant, identifier-legal string).
+
+```javascript
+// Bad 
+console.log(myObj['prop'])
+
+// Good
+console.log(myObj.prop)
+``` 
+
+## Use ES6 `import` rather than `require`
+
+ES6 provides a module syntax that we can use rather than a `require` function. Babel's implementation of the ES6 module loader uses CommonJS under the hood, so we can `import` external CommonJS modules with no problem.
+
+```javascript
+// Bad
+const Moment = require('moment')
+const MyModule = require('./my-module')
+
+// Good
+
+import Moment from 'moment'
+import * as MyModule from './my-module'
```

Comment:
I don't think so. Sometimes you want everything. Like in a reducer, you may import every action and have a reducer for each action.